diff -Naur a/google/pubsub/v1/BUILD.bazel b/google/pubsub/v1/BUILD.bazel
--- a/google/pubsub/v1/BUILD.bazel	2019-06-17 01:46:27.082195913 +0000
+++ b/google/pubsub/v1/BUILD.bazel	2019-06-17 01:55:29.647285414 +0000
@@ -149,3 +149,24 @@
         ":pubsub_go_proto",
     ],
 )
+
+##############################################################################
+# C++
+##############################################################################
+load(
+    "@com_google_googleapis_imports//:imports.bzl",
+    "cc_grpc_library",
+    "cc_proto_library",
+)
+
+cc_proto_library(
+    name = "pubsub_cc_proto",
+    deps = [":pubsub_proto"],
+)
+
+cc_grpc_library(
+    name = "pubsub_cc_grpc",
+    srcs = [":pubsub_proto"],
+    deps = [":pubsub_cc_proto"],
+    grpc_only = True,
+)
