diff -Naur a/google/pubsub/v1/BUILD.bazel b/google/pubsub/v1/BUILD.bazel
--- a/google/pubsub/v1/BUILD.bazel	2019-07-01 15:09:37.348895248 -0700
+++ b/google/pubsub/v1/BUILD.bazel	2019-07-01 15:12:29.851407500 -0700
@@ -143,3 +143,24 @@
         ":pubsub_go_proto",
     ],
 )
+
+##############################################################################
+# C++
+##############################################################################
+load(
+    "@com_google_googleapis_imports//:imports.bzl",
+    "cc_grpc_library",
+    "cc_proto_library",
+)
+
+cc_proto_library(
+    name = "pubsub_cc_proto",
+    deps = [":pubsub_proto"],
+)
+
+cc_grpc_library(
+    name = "pubsub_cc_grpc",
+    srcs = [":pubsub_proto"],
+    deps = [":pubsub_cc_proto"],
+    grpc_only = True,
+)

diff --git a/google/cloud/bigquery/storage/v1beta1/BUILD.bazel b/google/cloud/bigquery/storage/v1beta1/BUILD.bazel
index 1772fa579..cd46aa449 100644
--- a/google/cloud/bigquery/storage/v1beta1/BUILD.bazel
+++ b/google/cloud/bigquery/storage/v1beta1/BUILD.bazel
@@ -10,11 +10,13 @@ proto_library(
     name = "bigquerystorage_proto",
     srcs = [
         "avro.proto",
+        "arrow.proto",
         "read_options.proto",
         "storage.proto",
         "table_reference.proto",
     ],
     deps = [
+        "//google/api:client_proto",
         "//google/api:annotations_proto",
         "//google/api:resource_proto",
         "@com_google_protobuf//:empty_proto",
@@ -146,3 +148,24 @@ go_gapic_assembly_pkg(
         ":bigquerystorage_go_proto",
     ],
 )
+
+##############################################################################
+# C++
+##############################################################################
+load(
+    "@com_google_googleapis_imports//:imports.bzl",
+    "cc_grpc_library",
+    "cc_proto_library",
+)
+
+cc_proto_library(
+    name = "bigquerystorage_cc_proto",
+    deps = [":bigquerystorage_proto"],
+)
+
+cc_grpc_library(
+    name = "bigquerystorage_cc_grpc",
+    srcs = [":bigquerystorage_proto"],
+    deps = [":bigquerystorage_cc_proto"],
+    grpc_only = True,
+)

