diff -Naur a/third_party/py/python_configure.bzl b/third_party/py/python_configure.bzl
--- a/third_party/py/python_configure.bzl	2020-01-12 00:21:43.993457406 +0000
+++ b/third_party/py/python_configure.bzl	2020-01-12 00:22:35.084626229 +0000
@@ -307,13 +307,13 @@
 
 def _python_autoconf_impl(repository_ctx):
     """Implementation of the python_autoconf repository rule."""
-    _create_single_version_package(
-        repository_ctx,
-        "_python2",
-        _PYTHON2_BIN_PATH,
-        "python",
-        _PYTHON2_LIB_PATH,
-    )
+    #_create_single_version_package(
+    #    repository_ctx,
+    #    "_python2",
+    #    _PYTHON2_BIN_PATH,
+    #    "python",
+    #    _PYTHON2_LIB_PATH,
+    #)
     _create_single_version_package(
         repository_ctx,
         "_python3",
