diff -Naur a/bazel/cc_grpc_library.bzl b/bazel/cc_grpc_library.bzl
--- a/bazel/cc_grpc_library.bzl	2019-10-24 18:22:38.817788818 +0000
+++ b/bazel/cc_grpc_library.bzl	2019-10-24 18:23:25.991293690 +0000
@@ -101,5 +101,6 @@
             deps = deps +
                    extra_deps +
                    ["@com_github_grpc_grpc//:grpc++_codegen_proto"],
+            linkstatic = True,
             **kwargs
         )
diff -Naur a/bazel/python_rules.bzl b/bazel/python_rules.bzl
--- a/bazel/python_rules.bzl	2019-10-24 18:22:38.817788818 +0000
+++ b/bazel/python_rules.bzl	2019-10-24 18:25:31.128668957 +0000
@@ -175,7 +175,7 @@
         srcs = [
             ":{}".format(codegen_grpc_target),
         ],
-        deps = [Label("//src/python/grpcio/grpc:grpcio")] + deps,
+        #deps = [Label("//src/python/grpcio/grpc:grpcio")] + deps,
         **kwargs
     )
 
