diff -Naur a/include/ogg/os_types.h b/include/ogg/os_types.h
--- a/include/ogg/os_types.h	2019-12-03 16:33:10.000000000 -0800
+++ b/include/ogg/os_types.h	2019-12-03 16:33:42.000000000 -0800
@@ -70,6 +70,7 @@
 
 #elif (defined(__APPLE__) && defined(__MACH__)) /* MacOS X Framework build */
 
+#  include <stdint.h>
 #  include <sys/types.h>
    typedef int16_t ogg_int16_t;
    typedef uint16_t ogg_uint16_t;
