diff -Naur a/go/private/repositories.bzl b/go/private/repositories.bzl
--- a/go/private/repositories.bzl	2019-06-12 15:33:29.000000000 +0000
+++ b/go/private/repositories.bzl	2019-06-17 04:22:24.220740365 +0000
@@ -84,22 +84,6 @@
         commit = "582743bf40c5d3639a70f98f183914a2c0cd0680",  # v3.7.0, as of 2019-03-03
         shallow_since = "1551387314 -0800",
     )
-    # Workaround for protocolbuffers/protobuf#5472
-    # At master, they provide a macro that creates this dependency. We can't
-    # load it from here though.
-    if "net_zlib" not in native.existing_rules():
-        native.bind(
-            name = "zlib",
-            actual = "@net_zlib//:zlib",
-        )
-        http_archive(
-            name = "net_zlib",
-            build_file = "@com_google_protobuf//:third_party/zlib.BUILD",
-            sha256 = "c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1",
-            strip_prefix = "zlib-1.2.11",
-            urls = ["https://zlib.net/zlib-1.2.11.tar.gz"],
-        )
-
     _maybe(
         git_repository,
         name = "com_github_mwitkow_go_proto_validators",
